tool-openssl/crl_test.cc

```diff
@@ -0,0 +1,181 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "openssl/x509.h"
+#include <gtest/gtest.h>
+#include <openssl/pem.h>
+#include "internal.h"
+#include "test_util.h"
+#include "../crypto/test/test_util.h"
+#include <cctype>
+
+static X509_CRL* createTestCRL() {
+  X509_CRL *crl = X509_CRL_new();
+  if (!crl) {
+    ERR_print_errors_fp(stderr);
+    return nullptr;
+  }
+
+  // Set issuer name
+  X509_NAME *issuer = X509_NAME_new();
+  X509_NAME_add_entry_by_txt(issuer, "CN", MBSTRING_ASC, (unsigned char *)"Test CA", -1, -1, 0);
+  X509_CRL_set_issuer_name(crl, issuer);
+
+  // Set times
+  ASN1_TIME *lastUpdate = ASN1_TIME_new();
+  ASN1_TIME *nextUpdate = ASN1_TIME_new();
+  X509_gmtime_adj(lastUpdate, 0);
+  X509_gmtime_adj(nextUpdate, 86400L);  // 24 hours from now
+  X509_CRL_set1_lastUpdate(crl, lastUpdate);
+  X509_CRL_set1_nextUpdate(crl, nextUpdate);
+
+  // Add a revoked certificate
+  X509_REVOKED *revoked = X509_REVOKED_new();
+  ASN1_INTEGER *serialNumber = ASN1_INTEGER_new();
+  ASN1_INTEGER_set(serialNumber, 1);  // Serial number of revoked cert
+  X509_REVOKED_set_serialNumber(revoked, serialNumber);
+  X509_REVOKED_set_revocationDate(revoked, lastUpdate);
+  X509_CRL_add0_revoked(crl, revoked);
+
+  // Generate a key pair for signing
+  EVP_PKEY *pkey = EVP_PKEY_new();
+  RSA *rsa = RSA_generate_key(2048, RSA_F4, NULL, NULL);
+  EVP_PKEY_assign_RSA(pkey, rsa);
+
+  // Sign the CRL
+  X509_CRL_sign(crl, pkey, EVP_sha256());
```

Comment:
Best practice would be to check the return values of each function call to verify that something hasn't gone wrong.

I may be misremembering, but the GTEST macros might not be usable in a non-void function. We can still check the values similar to what this test function is doing:
https://github.com/aws/aws-lc/blob/096f73a62c3485bdb9feffe97eb0508fd8a89a03/tool-openssl/x509_test.cc#L12-L42